QGIS_plugin/bgtinlooptool/BGTInloopTool.py

```diff
@@ -255,7 +346,176 @@ def add_to_layer_group(
                 project.addMapLayer(qgs_lyr, addToLegend=False)
                 layer_group.insertLayer(0, qgs_lyr)
                 qgs_lyr.loadNamedStyle(qml)
-                qgs_lyr.triggerRepaint()
+                qgs_lyr.triggerRepaint()    
+ 
+    def gpkg_to_layer_group(self,gpkg_path: str, gpkg_layer_name: str, layer_group):
+        # Construct the data source URI for the GeoPackage
+        uri = f"{gpkg_path}|layername={gpkg_layer_name}"
+        
+        # Create a QgsVectorLayer object
+        qgs_lyr = QgsVectorLayer(uri, gpkg_layer_name, "ogr")
+        
+        if qgs_lyr.isValid():
+            project = QgsProject.instance()
+            project.addMapLayer(qgs_lyr, addToLegend=False)
+            layer_group.insertLayer(0, qgs_lyr)
+        else:
+            print(f"Failed to load layer '{gpkg_layer_name}' from '{gpkg_path}'.")
+
+class NetworkTask(QgsTask):
+    def __init__(self, url, output_gpkg,extent_bbox,extent_geometry_wkt,layer_name):
+        super().__init__("Download and Convert Data")
+        self.url = url
+        self.output_gpkg = output_gpkg
+        self.extent_bbox = extent_bbox
```

Comment:
Deze is raar. Je stopt er een boundingbox in maar vervolgens roep je wkt_to_bbox aan op regel 377. Je functie "wkt_to_bbox" moet dan anders heten want je doet niets met wkt.